### ISSUE TITLE
[discussion] "I understand that I will be blocked if I remove or skip any mandatory field"

### ISSUE THREAD

**Issue by chrizilla** [yt-dlp/yt-dlp#5758]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

Since [discussions](https://github.com/yt-dlp/yt-dlp/discussions) have been [disabled](https://github.com/yt-dlp/yt-dlp/discussions/685), I understand we are expected [to open an issue instead of a discussion](https://github.com/yt-dlp/yt-dlp/discussions/685), so here we go:

> "I understand that I will be blocked if I remove or skip any mandatory field"

I believe this new punishment of blocking users simply because they made an error in creating an issue is WAY to harsh. Mistakes happen. We all do our best here, but for most of us, it's simply not possible to spend hours searching the entire bugtracker, reading the entire issue guidelines, etc. before EVERY SINGLE issue we report. We have jobs, family, etc.

I think it's ok to expect all of this generally speaking, but it's simply impossible to avoid that once in a while something is overlooked. Again, mistakes happen. Sometimes I come across a bug or something that I think you will profit by knowing but don't have an hour to do ALL the checks (each and every time), so I have to decide if I keep it to myself or let you know anyway. I choose the latter. If you think that banning me for that is the best approach, please tell me, so that at least I know what risks I take.

I believe I am generally well informed, update yt-dlp regularly, have read the README and the guidelines multiple times and also do search the bugtracker for similar issues, but all of this is really complex (depending on the user's own skills) and I don't think it is feasible for regular contributors to do each of these steps each and every time. Also, it's possible to overlook something or forget something ...

This whole projects lives largely due to the valuable contributions from an entire community. I deem it really counterproductive to threaten good faith efforts with bans. It wasn't necessary and worked before this ban was introduced recently, so I strongly suggest to go back to the good old days and do away with threatening to ban the very contributors this project depends on.

I understand that it's frustrating to deal with imperfect issues, but you can't expect a first-time contributor to have the same perfection as you have after working on this program for years. You have created issue templates, guidelines, readmes, etc. so you have done your part to raise the quality level of contributions as much as possible. But after a certain point, perfectionism turns into pedantry which won't improve contribution quality. Quite on the contrary, it will turn away users (even by force if you ban them). So I strongly advise to reverse course before it comes to that.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by bashonly** [comment]
> We have jobs, family, etc.

^ this applies to the maintainers and devs as well though

I don't think the checklist is that tall of an ask. It's not telling you to memorize the entire README, it's asking if you have looked at it. Maybe CTRL+F for some keywords pertaining to your question. It's asking if you've searched the tracker, which IMO is common courtesy to do if you're about to ask some real live humans to help you with your problem. It's confirming that you're using the appropriate submission form and using the current version of the program. And it's asking that you submit complete-enough information (e.g. verbose log) so that something can actually be done about your question/problem/request. This all takes what, 5, 10 minutes tops?

Sure, that first checkbox may be a little confrontational. But without it, many more people WILL remove and skip mandatory fields.

Bottom line:

More low-effort/incomplete reports and duplicate/invalid issues = Less actual productivity and development

**Comment by chrizilla** [comment]
I think we are in agreement here, but

> Sure, that first checkbox may be a little confrontational

this is why I advise against it. There is no need to be confrontational. We are all doing our best.

**Comment by pukkandan** [comment]
> We have jobs, family, etc.

So do I

> > "I understand that I will be blocked if I remove or skip any mandatory field"
> 
> I believe this new punishment of blocking users simply because they made an error in creating an issue is WAY to harsh

> Mistakes happen.

I have never blocked anyone (or threatened to do so) for anything that can reasonably be called a mistake. Even in cases where the OP has clearly not even looked at pinned issues, we still reply before closing the issue. But what do you expect us to do when people intentionally mock us by deleting/editing the issue template - as if doing so gives them immunity from the requirements? [This is the lengths some people go to](https://github.com/yt-dlp/yt-dlp/issues?q=is%3Aissue+author%3Aupintheairsheep2+sort%3Aupdated-desc) for me to actually end up blocking them

> We all do our best here, 

I don't doubt you are, but [clearly not everyone](https://github.com/yt-dlp/yt-dlp/issues?q=is%3Aissue+sort%3Aupdated-desc+label%3Aincomplete%2Cinvalid). Like it or not, there has been a noticeably lesser number of zero effort issues since the warning was added, irrespective of whether I actually enforce it or not.

> it's simply not possible to spend hours searching the entire bugtracker

Instead maintainers should spend hours triaging them? Nobody expects you to "spend hours searching the bugtracker". All you are expected to do is a basic search, most likely just the site name or error message. That few minutes of your time accumulates to save hours of the maintainers' time. One user may be opening < 1 issue a month, but there was ~270 total issues opened in just last month. I get that github search sucks, which is why maintainers always link to the actual duplicate issue (unless it's already pinned) before closing them.

> It wasn't necessary and worked before this ban was introduced recently, so I strongly suggest to go back to the good old days and do away with threatening to ban the very contributors this project depends on.

The good old days when the project wasn't as popular and it was feasible for me to personally ask each user for the verbose log? That time is never coming back. Or do u mean the days when we were being bombarded with issues that has no effort put into it and blatantly disregards the template? Would you be willing to triage these issues going forward? If anyone is actually willing to take on that responsibility, I'll gladly remove the warning. If not, it's staying up

The only other option I see is to silently ignore issues that I don't consider worth my time and let the issue tracker become a mess of untriaged, duplicated, and invalid issues. Would that be preferable?

> so I have to decide if I keep it to myself or let you know anyway.

If it's not important enough for the user to spend some time writing a proper issue, why should devs spend any time trying to fix it? In that case, it may be better to not report it.

**Comment by dirkf** [comment]
Does GH provide a way with the new template system to trigger the issue search, ideally with `-label:duplicate`? This seems to be a feature in other bug trackers. If the user actually includes the backtrace, the error string and the name of the failing module very often identify the problem uniquely.

**Comment by pukkandan** [comment]
Nope. I've seen some other repos use bots that do this though (after issue creation)

**Comment by pukkandan** [comment]
Their continued comments in https://github.com/yt-dlp/yt-dlp/issues/6802 prove otherwise. But even if I did block someone, I am not interested in trying to justify my decision to everyone who asks.
